src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,29 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        Rectangle current = (Rectangle) obj;
+        return (width == null && current.width == null || width.equals(current.width))
+                && (length == null && current.length == null
+                    || length.equals(current.length))
+                && (color == null && current.color == null
+                    || color.equals(current.color));
```

Comment:
what if `width` is null and `current.width` is not null? what will we get on line 35?